### ISSUE TITLE
Category logging & reporting improvements

### ISSUE THREAD

**Issue by JohnGillanders** [coopdevs/timeoverflow#702]
This change request has two parts:
1. Allow manual choice of 'Category’ (same list as for offers/requests) when logging a time transfer not linked to a post. I.e. under the 'post' menu, there should be a 'None' option which then allows choice of a Category manually.
2. Show ‘Category’ as well as ‘Comments’ in the ’Transfers’ report. If the transfer is linked to an offer/request, the report should show the Category of the post, if not linked to an offer/request then it should show the Category manually chosen (if any) during the time transfer.

**Comment by JohnGillanders** [comment]
Addition to the original request:
When logging a time transfer, in the 'post' drop-down, it shows offers of the other person (the one who provided the service). Please also show requests of the current user (the person who received the work) and allow one of those to be chosen instead. Some timebanks mostly work from requests rather than offers, so this would be very helpful.

**Comment by JohnGillanders** [comment]
@sseerrggii @markets - some of the New Zealand timebanks want to raise some funds to pay for this development. Are you or someone there able to provide a quote to make these changes? Note I have added a third part to this request (edited the original description). If they have a clear quote, they can turn to their funders and request the money but they need the quote to be able to do that.
Thanks,
John

**Comment by sseerrggii** [comment]
Hi @JohnGillanders !! Your improvements proposal seems good to me, let me talk with the team to see when it's possible to do it

**Comment by JohnGillanders** [comment]
Hi Sergi,

How did you go with the suggested changes to TimeOverflow, did you have a chance to discuss it with the other developers?

Ngā mihi,
John Gillanders
Managing Director
Technaturally
www.Technaturally.info <http://www.technaturally.info/>
Your Life, Enriched by Technology
027 486 1533

**Comment by JohnGillanders** [comment]
Hi Sergi,Just checking, did you manage to discuss with the other developers?Regards,John GillandersManaging DirectorTechnaturallywww.Technaturally.infoYour Life, Enriched by Technology027 486 1533On 14/10/2023, at 8:50 PM, John Gillanders ***@***.***> wrote:﻿Hi Sergi,How did you go with the suggested changes to TimeOverflow, did you have a chance to discuss it with the other developers?
Ngā mihi,John GillandersManaging DirectorTechnaturallywww.Technaturally.infoYour Life, Enriched by Technology027 486 1533

On 25/09/2023, at 9:33 PM, Sergi Alonso ***@***.***> wrote:Hi @JohnGillanders !! Your improvements proposal seems good to me, let me talk with the team to see when it's possible to do it—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by sseerrggii** [comment]
Hi @JohnGillanders yes, but no one has answered me yet.

**Comment by markets** [comment]
Hi @JohnGillanders! Sorry my late reply, I was out for some weeks in my parental leave 👶🏼 But now we're back 💪🏼 

As @sseerrggii agreed with this development, I think we can start evaluating the proposed changes. Let me make an initial analysis and if I don't see impediments, I'll send you a quote and the schedule estimation to have it in production.

**Comment by markets** [comment]
@JohnGillanders @sseerrggii I made a quick investigation, here are my thoughts:
- 1 and 2: both seem quite feasible ✅ I have an idea about the implementation
- 3: if I understood this well, the idea is to allow transfers where de source and destination are the same account? This seems like a dangerous thing in the system no? For the sake of reference, it was fixed in 2016 in #231.

**Comment by JohnGillanders** [comment]
﻿Hi Marc,Congratulations on your new baby! Hope all is going well for you.Thanks for your follow up.To clarify for #3, no, that is not the intention.The hours are still being transferred from one person to another, but at the moment the person who received the service (who is ‘sending’ the hours) can only choose from posts which are offers of the other person. They should be able to choose to link one of their own request (‘inquiry’) posts to the transfer instead (but they are still sending the hours to the other person, who fulfilled their request/inquiry.Does that make sense?So e.g.:I make a request/inquiry post for ‘Software development’.You fulfill the request taking 3 hours.I transfer the hours to you and link my request/inquiry post to the transfer.One Timebank here in Aotearoa New Zealand does almost all transactions based on ‘request’ posts and hardly any on ‘offer’ posts but they cannot link the time transfers to the ‘requests’.Regards,John GillandersManaging DirectorTechnaturallywww.Technaturally.infoYour Life, Enriched by Technology027 486 1533On 1/11/2023, at 8:53 AM, Marc Anguera ***@***.***> wrote:﻿
@JohnGillanders @sseerrggii I made a quick investigation, here are my thoughts:

1 and 2: both seem quite feasible ✅ I have an idea about the implementation
3: if I understood this well, the idea is to allow transfers where de source and destination are the same account? This seems like a dangerous thing in the system no? For the sake of reference, it was fixed in 2016 in #231.

**Comment by sseerrggii** [comment]
Yes, for me is fine. You pay services related to offers to the receiver of the transfer but also is pretty common to pay because someone saw your requests and help with them. Now if you have this use case you don't relate this transfer with any post and with this feature we will solve this :)

**Comment by markets** [comment]
Ok! let's do it 💪🏼 

@JohnGillanders We'll start to work on that soon! Regarding the budget, not sure how many hours it will take the whole process (dev + testing + release), but I'd like to propose something similar like you did last time. Let me know if that works for you.

Thanks!

**Comment by JohnGillanders** [comment]
Hi Marc,Thank you. Do you mean we pay a fixed amount as a contribution for the development regardless of the actual number of hours?Please be aware that the TimeBanks will be applying to funders for grants to cover the cost of this and they might not yet have the budget for it. In fact we can’t guarantee that they will achieve the funding as it depends on the decision of the finders, and it’s also unclear how long funding would take. So you might want to hold off on the development until all that is clear…?The other factor is they also had two other separate requests which I also logged in GitHub here:Statistics->Global activity excluding 'TimeBank' account transactions · Issue #701 · coopdevs/timeoverflowgithub.comAnd here:Ability to edit/reassign tags in bulk · Issue #704 · coopdevs/timeoverflowgithub.comWould you please also evaluate these and let me know what you think - either as one overall cost for all three developments or even better separate costs representing the different efforts (bear in mind that some requests come from some TimeBanks and not from others so some may be willing to pay for one development but maybe not the other).Thank you.Regards,John GillandersManaging DirectorTechnaturallywww.Technaturally.infoYour Life, Enriched by Technology027 486 1533On 3/11/2023, at 7:00 AM, Marc Anguera ***@***.***> wrote:﻿
Ok! let's do it 💪🏼
@JohnGillanders We'll start to work on that soon! Regarding the budget, not sure how many hours it will take the whole process (dev + testing + release), but I'd like to propose something similar like you did last time. Let me know if that works for you.
Thanks!

**Comment by markets** [comment]
ok @JohnGillanders Let's continue this conversation via email, so we can keep this tracker for bugs, requests, ...

I'll send you an estimation of hours/effort for each request so it will be clearer for funders to decide.

Regards,
Marc
